test/FilterTest.cpp

```diff
@@ -18,10 +18,10 @@ namespace {
 ValueId asBool(bool value) { return Id::makeFromBool(value); }
 
 // Convert a generator to a vector for easier comparison in assertions
-std::vector<IdTable> toVector(cppcoro::generator<IdTable> generator) {
+std::vector<IdTable> toVector(Result::Generator generator) {
   std::vector<IdTable> result;
-  for (auto& table : generator) {
-    result.push_back(std::move(table));
+  for (auto& pair : generator) {
+    result.push_back(std::move(pair.idTable_));
```

Comment:
At least add a big comment that these tests will have lifetime issues if the input contains local vocab entries.